test_single_key fails on big-endian systems

```
FAIL: test_single_key (kitty_tests.datatypes.TestDataTypes)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/<<PKGBUILDDIR>>/kitty/launcher/../../kitty_tests/datatypes.py", line 556, in test_single_key
    self.assertNotEqual(hash(SingleKey(key=1, mods=2)), hash(SingleKey(key=1)))
    GLFW_MOD_KITTY = 256
    GLFW_MOD_SHIFT = 1
    SingleKey = <class 'fast_data_types.SingleKey'>
    m = 1
    s = SingleKey(mods=1)
    self = <kitty_tests.datatypes.TestDataTypes testMethod=test_single_key>
AssertionError: 1 == 1
```

Encountered on Debian's [hppa](https://buildd.debian.org/status/fetch.php?pkg=kitty&arch=hppa&ver=0.26.5-1&stamp=1671652623&raw=0) and [powerpc](https://buildd.debian.org/status/fetch.php?pkg=kitty&arch=powerpc&ver=0.26.5-1&stamp=1671651634&raw=0) builds.

--- a/kitty/keys.c
+++ b/kitty/keys.c
@@ -301,9 +301,17 @@ typedef uint64_t keybitfield;
 #endif
 typedef union Key {
     struct {
+#if __BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__
         keybitfield mods : MOD_BITS;
         keybitfield is_native: 1;
         keybitfield key : KEY_BITS;
+#elif __BYTE_ORDER__ == __ORDER_BIG_ENDIAN__
+        keybitfield key : KEY_BITS;
+        keybitfield is_native: 1;
+        keybitfield mods : MOD_BITS;
+#else
+#error "Unsupported endianness"
+#endif
     };
     keybitfield val;
 } Key;

